src/main/java/bio/terra/cli/app/DockerToolsManager.java

```diff
@@ -250,55 +257,74 @@ private Integer waitForDockerContainerToExit(String containerId) {
     return exec.awaitStatusCode();
   }
 
-  /** Read the Docker container logs into a string. TODO: How to handle very long output? */
-  private String getLogsForDockerContainer(String containerId) {
-    try {
-      return dockerClient
-          .logContainerCmd(containerId)
-          .withStdOut(true)
-          .withStdErr(true)
-          .exec(new DockerToolsManager.LogContainerTestCallback())
-          .awaitCompletion()
-          .toString();
-    } catch (InterruptedException intEx) {
-      logger.error("Error reading logs for Docker container.", intEx);
-      return "<ERROR READING CONTAINER LOGS>";
-    }
-  }
-
-  /** Delete the Docker container. */
+  /**
+   * Delete the Docker container.
+   *
+   * @param containerId id of the container
+   */
   private void deleteDockerContainer(String containerId) {
     dockerClient.removeContainerCmd(containerId).exec();
   }
 
+  /**
+   * Read the Docker container logs and write them to standard out.
+   *
+   * @param containerId id of the container
+   */
+  private void outputLogsForDockerContainer(String containerId) {
+    dockerClient
+        .logContainerCmd(containerId)
+        .withStdOut(true)
+        .withStdErr(true)
+        // .withTimestamps(true)
+        .withFollowStream(true)
+        .withTailAll()
+        .exec(new DockerToolsManager.LogContainerTestCallback());
+  }
+
   /** Helper class for reading Docker container logs into a string. */
   private static class LogContainerTestCallback extends ResultCallback.Adapter<Frame> {
 
     protected final StringBuffer log = new StringBuffer();
-    List<Frame> collectedFrames = new ArrayList<>();
-    boolean collectFrames = false;
+    List<Frame> framesList = new ArrayList<>();
+
+    boolean buildSingleStringOutput;
+    boolean buildFramesList;
 
     public LogContainerTestCallback() {
-      this(false);
+      this(false, false);
     }
 
-    public LogContainerTestCallback(boolean collectFrames) {
-      this.collectFrames = collectFrames;
+    public LogContainerTestCallback(boolean buildSingleStringOutput, boolean buildFramesList) {
+      this.buildSingleStringOutput = buildSingleStringOutput;
+      this.buildFramesList = buildFramesList;
     }
 
     @Override
     public void onNext(Frame frame) {
-      if (collectFrames) collectedFrames.add(frame);
-      log.append(new String(frame.getPayload(), Charset.forName("UTF-8")));
+      String logStr = new String(frame.getPayload(), Charset.forName("UTF-8"));
+
+      // TODO: PF-423. Calling sysout.println here breaks the model of printing user-facing output
+      // from the command classes only.
+      // Revisit this once we have better model for centralizing output across all commands/rest of
+      // the codebase.
```

Comment:
no action on this pr: one thing we could do is have our own interface to wrap system.out.println and have that be in charge with centralizing output. we would still have to allow printing output from outside of the command classes.

Reply:
Yes, that's along the lines of what I'm thinking for the follow-on ticket.